adala/runtimes/_litellm.py

```diff
@@ -37,32 +80,43 @@ class LiteLLMChatRuntime(LiteLLMInferenceSettings, Runtime):
 
     model_config = ConfigDict(arbitrary_types_allowed=True)  # for @computed_field
 
+    def as_inference_settings(self, **override_kwargs) -> LiteLLMInferenceSettings:
```

Comment:
I don't love this either. My ideal solution would be to use composition instead of inheritance:

```
from litellm.types.completion import CompletionRequest  # also a pydantic model

class LiteLLMChatRuntime(LiteLLMInferenceSettings, Runtime):

    # set defaults here and merge w/ user input - temperature=0, seed=47, etc
    inference_params: CompletionRequest
    # or even just
    # inference_params: dict
```

here's the source:
```
Source:
class CompletionRequest(BaseModel):
    model: str
    messages: List[str] = []
    timeout: Optional[Union[float, int]] = None
    temperature: Optional[float] = None
    top_p: Optional[float] = None
    n: Optional[int] = None
    stream: Optional[bool] = None
    stop: Optional[dict] = None
    max_tokens: Optional[int] = None
    presence_penalty: Optional[float] = None
    frequency_penalty: Optional[float] = None
    logit_bias: Optional[dict] = None
    user: Optional[str] = None
    response_format: Optional[dict] = None
    seed: Optional[int] = None
    tools: Optional[List[str]] = None
    tool_choice: Optional[str] = None
    logprobs: Optional[bool] = None
    top_logprobs: Optional[int] = None
    deployment_id: Optional[str] = None
    functions: Optional[List[str]] = None
    function_call: Optional[str] = None
    base_url: Optional[str] = None
    api_version: Optional[str] = None
    api_key: Optional[str] = None
    model_list: Optional[List[str]] = None

    model_config = ConfigDict(protected_namespaces=(), extra="allow")
File:           ~/.cache/pypoetry/virtualenvs/adala-Kfg50PSz-py3.11/lib/python3.11/site-packages/litellm/types/completion.py
Type:           ModelMetaclass
```
it includes everything you could pass to `litellm.{a}completion`.

However, this breaks backwards compat, because users are currently able to `OpenAIChatRuntime(temperature=0.5)` at the root level of the model.

I don't want to include all the above params at root level, ideally only the ones that we set defaults for that are different from litellm's defaults, but I still want the full list to be available for the user to set somehow. See the problem?

Reply:
Ah, I see. I don't mind having a model for a `Request`, I just don't like encapsulating a subset of the attributes under a `Settings` object, fwiw.
It's not obvious to me that we have to take an OO approach to encoding this though, and if we do I think it warrants more discussion than comments in a PR.
For now, I don't mind leaving it but I really think we need to avoid investing more in this until we discuss how we want to structure this stuff in a holistic way.